itou/static/js/htmx_handlers.js

```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+(function () {
+  htmx.on("htmx:afterOnLoad", (e) => $(e.detail.elt.dataset.openModal).modal("show"));
+
+  function makeToast(title, body) {
+    const toast = document.createElement("div");
+    toast.classList.add("toast");
+    toast.setAttribute("aria-atomic", "true");
+    toast.dataset.autohide = false;
+
+    const header = document.createElement("div");
+    header.classList.add("toast-header", "bg-danger", "text-white");
+
+    const headerTextContainer = document.createElement("strong");
+    headerTextContainer.appendChild(document.createTextNode(title));
+    headerTextContainer.classList.add("mr-auto");
+    header.appendChild(headerTextContainer);
+
+    const headerCloseButton = document.createElement("button");
+    headerCloseButton.classList.add("ml-2", "mb-1", "close");
+    headerCloseButton.setAttribute("data-dismiss", "toast");
+    headerCloseButton.setAttribute("aria-label", "Close");
+
+    const headerCloseLabel = document.createElement("span");
+    headerCloseLabel.setAttribute("aria-hidden", "true");
+    headerCloseLabel.appendChild(document.createTextNode("×"));
+    headerCloseButton.appendChild(headerCloseLabel);
+    header.appendChild(headerCloseButton);
+    toast.appendChild(header);
+
+    const toastBody = document.createElement("div");
+    toastBody.classList.add("toast-body", "bg-white");
+    toastBody.appendChild(document.createTextNode(body));
+    toast.appendChild(toastBody);
+
+    document.getElementById("toast-container").appendChild(toast);
+    $(toast).toast("show");
+  }
+
+  htmx.on("htmx:responseError", (e) => {
+    makeToast(
+      "Erreur de traitement code ${e.detail.xhr.status}",
+      `Statut ${e.detail.xhr.statusText} :\n\n${e.detail.xhr.responseText}`
+    );
+  })
```

Comment:
Je limiterais peut-être la création automatique du toast seulement aux erreurs 5xx.

Les 4xx ayant régulièrement une sémantique différente suivant l'URL appelée, et vu que ce sont des erreurs client ça ne me semble pas déconnant que ce soit le client (JS via XHR) qui doivent les gérer.
Ça peux nous embêter pour le debug mais au pire la personne dira "Je clique ça marche pas" plutôt que "J'ai une erreur 404". Et si je ne m'abuse `responseText` va leur afficher le contenu de ce qui est récupéré donc possiblement le HTML de nos pages d'erreur, non ?

Reply:
Le “problème” est que le client est HTMX, et HTMX gère les erreurs en émettant `htmx:responseError`. https://htmx.org/docs/#requests
Il me paraît plus agréable d’indiquer que l’opération a échoué aux utilisateurs, plutôt que de les laisser patienter devant l’indicateur de chargement (ou pire, une interface où rien n’a changé).

Après, afficher le code HTML n’est effectivement pas optimal. J’ai simplifié, on affiche simplement “Erreur code 404 (Not Found)” dans une `alert`.